packages/ui/src/SmartResourcePropertyDisplay.tsx

```diff
@@ -0,0 +1,19 @@
+import { PropertyType } from '@medplum/core';
+import { Resource } from '@medplum/fhirtypes';
+import { evalFhirPath } from '@medplum/fhirpath';
+import React from 'react';
+import { ResourcePropertyDisplay } from '.';
+
+export interface SmartResourcePropertyDisplayProps {
+  resource: Resource;
+  path: string;
+}
+
+export function SmartResourcePropertyDisplay(props: SmartResourcePropertyDisplayProps): JSX.Element {
+  const value = evalFhirPath(props.path, props.resource);
+
+  if (value?.[0] && value.length === 1) {
+    return <ResourcePropertyDisplay value={value?.[0] || ''} propertyType={PropertyType.string} />;
+  }
+  return <></>;
```

Comment:
You can change the function signature to return `JSX.Element | null` and just return null here.